rbac/management/group/definer.py

```
@@ -234,7 +240,11 @@ def update_group_roles(group, roleset, tenant):
 def _roles_by_query_or_ids(roles_or_role_ids: Union[QuerySet[Role], list[str]]) -> QuerySet[Role]:
     if not isinstance(roles_or_role_ids, QuerySet):
         # If given an iterable of UUIDs, get the corresponding objects
-        return Role.objects.filter(uuid__in=roles_or_role_ids)
+        filtered_roles = Role.objects.filter(uuid__in=roles_or_role_ids)
```

This is within if statement, the else case is missing.

---

> So only when all of the roles are not available, error will be thrown?
Yesh! Only when the role is invalid/non-existent will an error be thrown! If there is a group of roles that includes both existent and non-existent, there will be no error thrown, because the non-existent role will not be added to the group.

---

> This is within if statement, the else case is missing.


[2025-06-17 18:44:39,473] ERROR: Error encountered during group seeding This role is nonexistent/nonvalid and cannot be added to the group.

^^^When I place a similar statement within the else case, there are errors right when it starts with the seeding. I was thinking that since when a role is being added to a group, only the UUID is valid through the body of data.

---

Could you investigate why this error will be thrown? Because it is complaining that roles added to a group all do not exist, I wonder how could it happen for a seeding process

---

So this sounds like what we need, right? throwing 404 if the list is empty. Maybe we should add it and update the tests?

---

That sounds good to me, and I can add it! 
I was a bit unsure of whether the behavior should be normalized for the seeding so that's why I was hesitant to add it :sweat

---

Ah, I see. Yea, I wonder if the seeding would encounter this issue because it is supposed to add roles that's all exist. Maybe better to throw an error in this case so we know it is unexpected instead of silently skip it 🤔

---

Do you think maybe throwing a warning would suffice? 
I think with throwing an error, for the unit tests, the issue also comes up when the roles are trying to be "set up". The tests start to fail whenever the tests are being set up for the tests relating to the cross_access/test_util.py